src/gd.c

```diff
@@ -2939,7 +2939,8 @@ BGD_DECLARE(gdImagePtr) gdImageClone (gdImagePtr src) {
 		dst->tileColorMap[i] = src->tileColorMap[i];
 	}
 
-	if (src->polyAllocated > 0) {
+	if (src->polyAllocated > 0 && overflow2(sizeof(int), src->polyAllocated) == 0) {
+		dst->polyInts = (int *) gdMalloc (sizeof (int) * src->polyAllocated);
```

Comment:
the `(int *)` cast isn't necessary.  i know we use it when setting up polyInts, but we should remove those.  not that you have to do that cleanup here.

Reply:
Thank you for confirming. As you pointed out, this cast isn't necessary, so I delete it.